boot/cmdline.go

```diff
@@ -98,6 +100,34 @@ func getBootloaderManagingItsAssets(where string, opts *bootloader.Options) (boo
 	return mbl, nil
 }
 
+func bootVarsForcommandLineFromGadget(gadgetSnap string) (map[string]string, error) {
+	sf, err := snapfile.Open(gadgetSnap)
+	if err != nil {
+		return nil, fmt.Errorf("cannot open gadget snap: %v", err)
+	}
+	extraOrFull, full, err := gadget.KernelCommandLineFromGadget(sf)
```

Comment:
Hmm, perhaps it should be made clearer in the comment that this is only applicable if you have trusted assets bootloader, which LK does not implement

Reply:
I've tweaked the name and added a comment that this is only useful for bootloader for which we manage the boot configs. It will probably be more visible with the next PR, that the call is only used if the bootloader is a TrustedAssetsBootloader. Does that sound ok?

Reply:
Hmm, so modifying the kernel command line this way will only be supported for grub essentially? I thought that this was going to be a generic feature for all gadget / bootloaders that want to support it and that i.e. boot.scr on the pi will be updated to read this boot var and append things there.

Reply:
Right now it's targeting grub, where we control the boot scripts, know how the resulting command line looks like and the default command line we have is inconvenient for some use cases, but needing to have secure boot enabled means one cannot arbitrarily change the boot script or command line in the gadget.  Other bootloaders can already freely override whatever defaults are present in gadgets published by us.